src/gigs-board/components/posts/PostEditor.jsx

```diff
@@ -208,12 +259,24 @@ const labelEditor = (
     <Typeahead
       multiple
       labelKey="name"
+      onInputChange={checkLabel}
       onChange={setLabels}
       options={existingLabels}
       placeholder="near.social, widget, NEP, standard, protocol, tool"
       selected={state.labels}
       positionFixed
-      allowNew
+      allowNew={(results, props) => {
+        return (
+          !existingLabelSet.has(props.text) &&
+          props.selected.filter((selected) => selected.name === props.text)
+            .length == 0 &&
+          Near.view(
+            nearDevGovGigsContractAccountId,
+            "is_allowed_to_use_labels",
+            { editor: context.accountId, labels: [props.text] }
+          )
```

Comment:
I feel like it may break since `Near.view` is not blocking, and on the first (non-cached) run it will return null/undefined. Do I miss something?

Reply:
Experiments show this allowNew is invoked twice. Try type a new label, it shows from "No matches found" (non-cached) to "New selection xxxx" (Near.view resolves). If type label xxxx again, cache is hit, it immediately shows "New selection xxx".